plugin/evm/atomic/mempool.go

```diff
@@ -253,6 +239,11 @@ func (m *Mempool) addTx(tx *atomic.Tx, force bool) error {
 	if _, exists := m.txHeap.Get(txID); exists {
 		return fmt.Errorf("%w: tx %s is pending", errTxAlreadyKnown, tx.ID())
 	}
+	if !local {
+		if _, exists := m.discardedTxs.Get(txID); exists {
+			return fmt.Errorf("%w: tx %s was discarded", errTxAlreadyKnown, tx.ID())
+		}
+	}
```

Comment:
restores the same behaviour to drop txs that already dropped in mempool(was in legacy sdk [here](https://github.com/ava-labs/coreth/blob/v0.14.0/plugin/evm/handler.go#L73))